dimes/dimes.py

```diff
@@ -131,19 +131,39 @@ def finalize_plot(self):
                                         axis.units,
                                     ),
                                     name=time_series.name,
-                                    yaxis=f"y{axis_id}",
+                                    yaxis=f"y{y_axis_id}",
+                                    xaxis=f"x{x_axis_id}",
                                     mode="lines",
                                     visible="legendonly" if not time_series.is_visible else True,
                                     line={
                                         "color": time_series.color,
                                         "dash": time_series.line_type,
                                     },
                                 ),
-                                row=None if number_of_subplots == 1 else subplot_number,
-                                col=None if number_of_subplots == 1 else 1,
                             )
-                        self.fig.layout[f"yaxis{axis_id}"]["title"] = axis.get_axis_label()
+                        is_base_y_axis = subplot_base_y_axis_id != y_axis_id
+                        self.figure.layout[f"yaxis{y_axis_id}"] = {
+                            "title": axis.get_axis_label(),
+                            "domain": subplot_domains[subplot_number - 1]
+                            if subplot_base_y_axis_id == y_axis_id
+                            else None,
+                            "side": y_axis_side,
+                            "anchor": f"x{x_axis_id}",
+                            "overlaying": f"y{subplot_base_y_axis_id}" if is_base_y_axis else None,
+                            "tickmode": "sync" if not is_base_y_axis else None,
+                            "autoshift": True if axis_number > 1 else None,
+                        }
                         absolute_axis_index += 1
+                        y_axis_side = "right" if y_axis_side == "left" else "left"
+                    is_last_subplot = subplot_number == number_of_subplots
+                    self.figure.layout[f"xaxis{x_axis_id}"] = {
```

Comment:
How did you figure out that you could specify a particular axis for figure.layout? Maybe this will help me with reviewing documentation?

Reply:
I got most of it from this page: https://plotly.com/python/multiple-axes/

I may have found a few other examples via Google search.

Reply:
Do the following two lines perform the same thing?

self.figure.layout[xaxis2] = {"title":"graph"}
self.figure.layout(xaxis2 = {"title":"graph"})